### PR TITLE
 UFS-SRW v3.0.0 SciDoc updates

### PR DESCRIPTION
**Pull request by mzhangw**
It is to initialize scientific documentation update for UFS-SRW v3.0.0 release from DTC.
The web link is [here](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/index.html).

### CODE REVIEW

**Comment by mkavulich** [comment]
I am worried about the size of these images; ideally there should be *no* binary files (images included) in a github repository, especially for one that is a submodule for many other repositories. 5MB may not seem like a lot but that is an extra 5MB every time the repository is cloned, and it increases the entire ccpp-physics repository size by ~15%. And if we continue adding images for every new capability this problem will only get worse.

Is there a way for us to host these images elsewhere instead of keeping them in the ccpp-physics repository?

**Comment by ligiabernardet** [comment]
@mzhangw Pls provide la link to the rendered html so we can use as part of the review. Tks

@mkavulich Good point. What do you suggest? I suppose it could be distributed as a release asset??

**Comment by ligiabernardet** [comment]
Thanks for collecting these updates.
Given how extensive the changes described in the .txt files are, is the inline documentation contained in the Fortran files still current or does it need to be updated?

- index.html
  - app -> App
  - Please add an explanation to say that the GFS v16 suite is not the same code as the operational GFS v16 code uses. First because GFS v16 does not use CCPP at all. Second because the code has marched ahead. Similarly, the code in the all other suites has evolved, so the RAP suite/param is not what the operational RAP uses. Then say that P8 is like P8A (not the final P8) but also with evolved schemes. I suggest repeating this info on the page for each suite.
- rap_suite_page.html: Currently, the RAP suite is supported in SCM only. (not correct)
- _r_r_f_s_v1beta_page.html
   - The RRFS_v1beta suite is the primary suite target for the upcoming operational implementation of the Rapid Refresh Forecast System (RRFS) (not correct. The operational RRFS v1 will use RUC LSM and may use GF. Say that this WAS the primary target but that things have evolved since then)
   -  which is used in the UFS SRW App -> which can be configured using the UFS SRW App
- HRRR and RAP suite: you can add that these are candidates for RRFS v1 operational implementation
- _wo_f_s_v0_page.html
    -  targeted for use in the upcoming operational implementation of the NOAA's Warn-on-Forecast System (WoFS) and for the RRFS ensemble. 
- c_l_m__l_a_k_e_model.html
   - Define acronyms at first occurrence
   -  **The** CLM lake model...
   -  It also computes 2-m temperature: Of the air?? That is surprising to me.
   - in the regional application of UFS (RRFS) -> in the UFS SRW App
   - To cold-start the CLM lake model in RRFS -. To cold-start the CLM lake model in the UFS SRW App
   - The differences of surface variables from the RRFS 6-h forecast -> The differences of surface variables from the experimental RRFS 6-h forecast
- RUC LSM
   - some of these recent modifications -> some of these modifications
   - The sensitivity of surface fluxes and turbine-height winds to the RUC LSM parameters has been explored by Geng Xia, NREL. This study will determine the uncertainty range for the selected parameters in the RUC LSM and will be described in the journal paper. -> The sensitivity of surface fluxes and turbine-height winds to the RUC LSM parameters has been explored by Geng Xia, NREL to determine the uncertainty range for the selected parameters in the RUC LSM.
   -NOAH LSM -> Noah LSM
- NoahMP LSM
   - It is not necessary to describe how to cite the reference. People can click on the link and go see how to cite. However, the citation in the CCPP SciDoc differs from what is recommended in the text. Should it be changed to be as shown in the "how to cite"?
- GF
   - In GF scale awareness, and the aerosol dependence through rain generation (following Berry (1968) [[28]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_berry_1968) and evaporation formulations (following Jiang et al. (2010) [[102]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_Jiang_2010) ), depending on the cloud concentration nuclei at cloud base were added. -> The GF scheme takes into account aerosol dependence through rain generation (following Berry (1968) [[28]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_berry_1968) and evaporation formulations (following Jiang et al. (2010) [[102]] (https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_Jiang_2010) ), depending on the cloud concentration nuclei at cloud base. (No need to mention scale awareness here- it is discussed later in this paragraph)
   - GF included -> The GF scheme includes
   - massflux -> mass flux
   - PDF's -> PDFs
   - The deep and congestus convection considers scale awareness (Arakawa et al. (2011) [[8]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_Arakawa_2011) ), the shallow convection is not scale-aware. -> The deep and congestus convection considers scale awareness (Arakawa et al. (2011) [[8]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_Arakawa_2011) ). However, the shallow convection is not scale-aware.
   - Aerosol dependence is implemented through dependence of rain generation and evaporation formulations depending on the cloud concentration nuclei at cloud base (Berry 1968 [[28]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_berry_1968), Jiang et al.(2010) [[102]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_Jiang_2010), and Lee and Feingold (2010) [[115]](https://dtcenter.ucar.edu/GMTB/UFS_SRW_App_v3.0.0p/sci_doc/citelist.html#CITEREF_lee_and_feingold_2010) ). Not needed??? Already mentioned previously in this paragraph??
   - Aerosol dependence is considered experimental and is turned off at this point. -> Aerosol dependence is considered experimental and is not supported in this release. This should be mentioned earlier in the paragraph, when aerosol-dependence is introduced.
   - CCPP v6.0.0. I recommend incorporating info related to all previous releases into the main text. Then just describe the changes for the current distribution (SRW v3)
   - Operational Impacts in RAP/HRRR -> Why is this under "Physics Updates"? The last RAP/HRRR op impl was a while back.
   - Updates for aerosol-awareness (I suggest removing this since it seems it is not supported in the App)

I may have more comments later.

**Comment by mkavulich** [comment]
@mzhangw Can you also include a change in `physics/docs/pdftxt/suite_input.nml.txt`? Currently the section for `imp_physics` shows the default value as 99 but does not describe what this default is, can you add its description (should be `Zhao-Carr/Sundqvist microphysics cloud` I believe)?

@ligiabernardet I am not 100% sure (the doxygen documentation is unclear), but it should be possible to point to external images when building documents with Doxygen, using the syntax`@image html https://example.com/image.jpg`. We could host these images by uploading them to a page on the [Github wiki](https://github.com/ufs-community/ccpp-physics/wiki); this is what we have done for hosting some large images for the UFS SRW App users guide.

Thanks!

**Comment by mkavulich** [comment]
@mzhangw I opened a PR to your branch (https://github.com/mzhangw/ccpp-physics/pull/7) with changes that would move the images to an external location. I did a test locally to make sure it would work the way I expected in doxygen and it looks like it does. Let me know if you have any questions.

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    adjust_dry_mass = .false.
+    bc_update_interval = 6
+    beta = 0.0
+    consv_am = .false.
+    consv_te = 0.0
+    d2_bg = 0.0
+    d2_bg_k1 = 0.2
+    d2_bg_k2 = 0.04
+    d4_bg = 0.12
+    d_con = 1.0
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:68]: In RRFS we use d_con=0.5

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+    external_ic = .true.
+    fill = .true.
+    full_zs_filter = .false.
+    fv_debug = .false.
+    fv_sg_adj = 300
+    gfs_phil = .false.
+    hord_dp = -5
+    hord_mt = 5
+    hord_tm = 5
+    hord_tr = 10
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:88]: RRFS uses:    hord_dp = 6
    hord_mt = 6
    hord_tm = 6
    hord_tr = 8
    hord_vt = 6

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    ke_bg = 0.0
+    kord_mt = 9
+    kord_tm = -9
+    kord_tr = 9
+    kord_wz = 9
+    layout = 5, 2
+    make_nh = .true.
+    mountain = .false.
+    n_split = 5
+    n_sponge = 24
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:102]: In RRFS:
n_sponge = 9

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+    n_zs_filter = 0
+    na_init = 1
+    ncep_ic = .false.
+    nggps_ic = .true.
+    no_dycore = .false.
+    nord = 3
+    nord_tr = 2
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:109]: In RRFS:
nord_tr=0

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+    npx = 220
+    npy = 132
+    npz = 64
+    nrows_blend = 10
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:113]: In RRFS nrows_blend = 20

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+    phys_hydrostatic = .false.
+    print_freq = 6
+    psm_bc = 1
+    range_warn = .true.
+    read_increment = .false.
+    regional = .true.
+    regional_bcs_from_gsi = .false.
+    res_latlon_dynamics = 'fv3_increment.nc'
+    reset_eta = .false.
+    rf_cutoff = 2000.0
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:127]: In RRFS:    rf_cutoff = 2000.0
                  **sg_cutoff = 10000.0**

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+    write_restart_with_bcs = .false.
+    z_tracer = .true.
+/
+
+&fv_grid_nml
+    grid_file = 'INPUT/grid_spec.nc'
+/
+
+!>[GFS_PHYSICS_NML]
+&gfs_physics_nml
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:144]: RRFS rfus with smoke. Add extra lines:    
    addsmoke_flag = 1
    aero_dir_fdb = .true.
    aero_ind_fdb = .false.

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+    cnvgwd             = .false.
+    cplflx             = .false.
+    debug              = .false.
+    do_deep            = .false.
+    do_gsl_drag_ls_bl  = .true.
+    do_gsl_drag_ss     = .true.
+    do_gsl_drag_tofd   = .true.
+    do_mynnedmf        = .true.
+    do_mynnsfclay      = .true.
+    dspheat            = .true.
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:160]: In RRFS add after this line:    do_plumerise = .true.
    do_tofd = .false.
    do_ugwp = .false.
    do_ugwp_v0 = .false.
    do_ugwp_v0_nst_only = .false.
    do_ugwp_v0_orog_only = .false.
    drydep_opt = 1
    do_plumerise = .true.
    do_tofd = .false.
    do_ugwp = .false.
    do_ugwp_v0 = .false.
    do_ugwp_v0_nst_only = .false.
    do_ugwp_v0_orog_only = .false.
    drydep_opt = 1

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+    effr_in            = .true.
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:161]: add after this line:    enh_mix = .true.

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+    fhcyc              = 0.0
+    fhlwr              = 1200.0
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:163]: In RRFS:    fhlwr = 900.0
    fhswr = 900.0

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+    fhswr              = 1200.0
+    fhzero             = 1.0
+    gwd_opt            = 3
+    h2o_phys           = .true.
+    hybedmf            = .false.
+    iaer               = 5111
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:169]: iaer = 1011

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+    ialb               = 1
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:170]: ialb=2

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+    iau_delthrs        = 6
+    iau_inc_files      = ''
+    iaufhrs            = 30
+    icliq_sw           = 2
+    icloud_bl          = 1
+    ico2               = 2
+    iems               = 1
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:177]: iems=2

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 14]
```diff
@@ -0,0 +1,295 @@
...
+    iopt_frz           = 1
+    iopt_inf           = 1
+    iopt_rad           = 1
+    iopt_run           = 1
+    iopt_sfc           = 1
+    iopt_snf           = 4
+    iopt_stc           = 1
+    iopt_tbot          = 2
+    iopt_trs           = 2
+    iovr               = 3
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:194]: After iovr:   
    isncond_opt = 2
    isncovr_opt = 3

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 15]
```diff
@@ -0,0 +1,295 @@
...
+    ivegsrc            = 1
+    kice               = 9
+    ldiag3d            = .false.
+    lheatstrg          = .false.
+    lradar             = .true.
+    lsm                = 3
+    lsoil              = 4
+    lsoil_lsm          = 9
+    ltaerosol          = .true.
+    lwhtr              = .true.
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:208]: Insert here:    
    mix_chem = .true.
    mosaic_lu = 1
    mosaic_soil = 1

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+    nsfullradar_diag   = 3600
+    nst_anl            = .true.
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:210]: nst lines could be removed

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 17]
```diff
@@ -0,0 +1,295 @@
...
+    nstf_name          = 2, 1, 0, 0, 0
+    oz_phys            = .false.
+    oz_phys_2015       = .true.
+    pdfcld             = .false.
+    pre_rad            = .false.
+    print_diff_pgr     = .false.
+    prslrd0            = 0.0
+    random_clds        = .false.
+    redrag             = .true.
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:219]: Add:    
    rrfs_sd = .true.
    rrfs_smoke_debug = .false.

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 18]
```diff
@@ -0,0 +1,295 @@
...
+    satmedmf           = .false.
+    sfclay_compute_flux = .true.
+    shal_cnv           = .false.
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:222]: add:
    smoke_forecast = 0

File: physics/docs/pdftxt/RE300/FV3_HRRR_input.nml [hunk 19]
```diff
@@ -0,0 +1,295 @@
...
+    swhtr              = .true.
+    trans_trac         = .true.
+    ttendlim           = -999
+    use_ufo            = .true.
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/FV3_HRRR_input.nml:226]: Add:     
wetdep_ls_opt = 1

File: physics/docs/pdftxt/RE300/namelists/input.nml.v21.FV3_HRRR [hunk 20]
```diff
@@ -0,0 +1,299 @@
...
+    max_files_w = 100
+/
+
+&fms_nml
+    clock_grain = 'ROUTINE'
+    domains_stack_size = 5000000
+    print_memory_usage = .false.
+/
+
+&fv_core_nml
```
Comment (tanyasmirnova) [inline physics/docs/pdftxt/RE300/namelists/input.nml.v21.FV3_HRRR:56]: same changes here as in the previous FV3_HRRR

**Review by mkavulich — CHANGES_REQUESTED** [review]
@mzhangw Can you please merge the changes to remove the large images from this PR? If you have any questions please let me know.

https://github.com/mzhangw/ccpp-physics/pull/7

**Comment by mkavulich** [comment]
Since Man is now on leave, we have re-opened this PR at #91 

@tanyasmirnova Thank you for your comments, I will reach out to you to make sure these get addressed.

**Pull request closed without merging**